modules/kubernetes-addons/opentelemetry-operator/main.tf

```diff
@@ -1,17 +1,33 @@
-module "operator" {
-  source        = "../helm-addon"
-  helm_config   = local.helm_config
-  irsa_config   = null
+module "cert_manager" {
+  source        = "../cert-manager"
   addon_context = var.addon_context
-
-  depends_on = [kubernetes_namespace_v1.prometheus]
 }
 
-resource "kubernetes_namespace_v1" "prometheus" {
+resource "kubernetes_namespace_v1" "adot" {
   metadata {
     name = local.helm_config["namespace"]
     labels = {
       "app.kubernetes.io/managed-by" = "terraform-aws-eks-blueprints"
+      "control-plane"                = "controller-manager"
     }
   }
 }
+
+data "kubectl_path_documents" "adot" {
+  pattern = "${path.module}/manifests/*.yaml"
+}
+
+# official kubernetes_manifest issue
+# https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs/resources/manifest#before-you-use-this-resource
+resource "kubectl_manifest" "adot" {
+  for_each   = toset(data.kubectl_path_documents.adot.documents)
+  yaml_body  = each.value
+  depends_on = [kubernetes_namespace_v1.adot]
```

Comment:
This is just an FYI - we generally want to avoid `depends_on` as much as possible. To get around this, look for the "breadcrumbs" that you can feed into a child module/resource to make it implicitly dependent on the parent module/resource. Deployment order is a core tenant of Terraform and providing that information to Terraform will ensure things are deployed at the optimal time without unintended side-effects (explicit dependencies will cause the child resource/module to re-compute all computed attributes when the parent resource/module is updated - this can lead to some really bad experiences and infra service disruptions)


1. Yes, as you've discovered the `kubernetes_manifest` resource has some gnarly limitations. However, we don't need to use that here - we can use the specific resource types like `kubernetes_cluster_role_v1`,  `kubernetes_cluster_role_binding_v1`, etc., or we can just merge the manifests into one file like the AWS docs recommend and provide the [link](https://amazon-eks.s3.amazonaws.com/docs/addons-otel-permissions.yaml) for and just do a single `kubectl_manifest` resource without the `for_each` loop. 
2. Going back to the point of "breadcrumbs", if we create a template variable in the manifest file for [`namespace`](https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/627/files#diff-0393764fcbfc5379a2df275af001d20d3e8c190ab4570efcb83340c3eeee8bc1R5), we could potentially get two wins - ensure the namespaces are synced across resources, ensure the proper deployment order (terraform will see this input from the `kubernetes_namespace_v1` resource and know it needs to be created first and we can drop the `depends_on`)